relay-general/src/store/normalize.rs

```diff
@@ -661,6 +663,41 @@ fn normalize_logentry(logentry: &mut Annotated<LogEntry>, _meta: &mut Meta) -> P
     logentry.apply(|le, meta| logentry::normalize_logentry(le, meta))
 }
 
+// Reads device specs (family, memory, cpu, etc) from context and sets the device.class tag to high, medium, or low.
+fn normalize_device_class(event: &mut Event) {
+    let tags = &mut event.tags.value_mut().get_or_insert_with(Tags::default).0;
+    let tag_name = "device.class".to_string();
+    if let Some(ref mut contexts) = event.contexts.value_mut() {
+        if let Some(Context::Device(ref mut device)) = contexts.get_context_mut("device") {
+            if let Some(family) = device.family.value() {
+                if family == "iPhone" || family == "iOS" || family == "iOS-Device" {
+                    if let Some(processor_frequency) = device.processor_frequency.value() {
+                        if processor_frequency < &2000 {
+                            tags.insert(tag_name, Annotated::new(DeviceClass::LOW.to_string()));
+                        } else if processor_frequency < &3000 {
+                            tags.insert(tag_name, Annotated::new(DeviceClass::MEDIUM.to_string()));
+                        } else {
+                            tags.insert(tag_name, Annotated::new(DeviceClass::HIGH.to_string()));
+                        }
+                    }
+                } else if let (Some(&freq), Some(&proc), Some(&mem)) = (
+                    device.processor_frequency.value(),
+                    device.processor_count.value(),
+                    device.memory_size.value(),
+                ) {
+                    if freq < 2000 || proc < 8 || mem < 4 * GIB {
+                        tags.insert(tag_name, Annotated::new(DeviceClass::LOW.to_string()));
+                    } else if freq < 2500 || mem < 6 * GIB {
+                        tags.insert(tag_name, Annotated::new(DeviceClass::MEDIUM.to_string()));
+                    } else {
+                        tags.insert(tag_name, Annotated::new(DeviceClass::HIGH.to_string()));
+                    }
+                }
+            }
+        }
```

Comment:
1. This potentially overwrites a user tag with the same name. Are we fine with that?
2. Will the product be able to handle cases where this function is unable to derive a device class, and a user tag with the same name exists? If not, I would delete the tag here if no device class could be derived.

Reply:
1. Yes, we are ok with overwriting user tags with the same name. This should only be a sentry provided tag.
2. 👍 Added a line to remove any user provided values for the tag in `normalize_device_class`